WcaOnRails/app/models/competition.rb

```diff
@@ -2011,4 +2011,28 @@ def dues_per_competitor_in_usd
     dues = DuesCalculator.dues_per_competitor_in_usd(self.country_iso2, self.base_entry_fee_lowest_denomination.to_i, self.currency_code)
     dues.present? ? dues : 0
   end
+
+  def dues_payer
```

Comment:
I am unhappy with the fact that this can either return a `WfcXeroUser` or a simple `User`.

Rails is a weakly typed language, which usually means that you have to pay attention to all the typing yourself (as there is no compiler that can do it for you). As a result, deliberately creating methods that can return multiple different types should be considered bad practice.

Is there any way you can redesign your approach? For example there can be a `xero_dues_payer` method that returns only `WfcXeroUser` or `nil`, and then for other checks you can verify whether that return value is present or not and act accordingly.

Reply:
I've done it in another way, does it make sense now?

Reply:
See my follow-up review